python/modules/IcePy/Util.cpp

```diff
@@ -954,8 +958,39 @@ IcePy::convertException(const Ice::Exception& ex)
             PyObject_SetAttrString(p.get(), STRCAST("unknown"), s.get());
         }
     }
-    catch (const Ice::Exception&)
+    catch (const Ice::Exception& e)
+    {
+        ostr << e;
+        string str = ostr.str();
+
+        type = lookupType("Ice.UnknownException");
+        assert(type);
+        p = createExceptionInstance(type);
+        if (p.get())
+        {
+            PyObjectHandle s = createString(str);
+            PyObject_SetAttrString(p.get(), STRCAST("unknown"), s.get());
+        }
+    }
+    catch (const std::exception& e)
     {
+        ostr << e.what();
+        string str = ostr.str();
+
+        type = lookupType("Ice.UnknownException");
+        assert(type);
+        p = createExceptionInstance(type);
+        if (p.get())
+        {
+            PyObjectHandle s = createString(str);
+            PyObject_SetAttrString(p.get(), STRCAST("unknown"), s.get());
```

Comment:
Cant we just pass the `const char*` returned by `e.what()` into createString?

```
PyObjectHandle s = createString(e.what());
```

Reply:
Yes, createString takes a string_view. Fixed.